test/tasks/karpenter-test-suite.yaml

```diff
@@ -0,0 +1,127 @@
+        --stack-name "$(params.cluster-name)" \
+        --template-file website/content/en/preview/getting-started/getting-started-with-eksctl/cloudformation.yaml \
+        --capabilities CAPABILITY_NAMED_IAM \
+        --parameter-overrides "ClusterName=$(params.cluster-name)"
+
+  - name: create-cluster-configuration
+    image: alpine/curl:3.14
+    workingDir: $(workspaces.ws.path)/$(params.repository)
+    script: |
+      cat <<EOF > cluster.yaml
+      ---
+      apiVersion: eksctl.io/v1alpha5
+      kind: ClusterConfig
+      metadata:
+        name: $(params.cluster-name)
+        region: $(cat $(results.region.path))
+        version: "1.22"
+        tags:
+          karpenter.sh/discovery: $(params.cluster-name)
+      managedNodeGroups:
+        - instanceType: m5.large
+          amiFamily: AmazonLinux2
+          name: $(params.cluster-name)-system-pool
+          desiredCapacity: 2
+          minSize: 2
+          maxSize: 2
+      iam:
+        withOIDC: true
+      EOF
+
+  - name: create-cluster
+    image: weaveworks/eksctl:v0.94.0
+    workingDir: $(workspaces.ws.path)/$(params.repository)
+    script: |
+      eksctl create cluster -f cluster.yaml || eksctl upgrade cluster -f cluster.yaml
+
+  - name: update-kubeconfig
+    image: amazon/aws-cli:2.7.11
+    script: |
+      aws eks update-kubeconfig --name $(params.cluster-name)
+      cat /root/.kube/config > $(results.kubeconfig.path)
+
+  - name: create-iam-identity-mapping
+    image: weaveworks/eksctl:v0.94.0
+    script: |
+      eksctl create iamidentitymapping \
+      --username system:node:{{EC2PrivateDNSName}} \
+      --cluster "$(params.cluster-name)" \
+      --arn "arn:aws:iam::$(cat $(results.region.account-id)):role/KarpenterNodeRole-$(params.cluster-name)" \
+      --group system:bootstrappers \
+      --group system:nodes
+
+  - name: create-karpenter-service-account
+    image: weaveworks/eksctl:v0.94.0
+    script: |
+      eksctl create iamserviceaccount \
+        --cluster "$(params.cluster-name)" --name karpenter --namespace karpenter \
+        --role-name "$(params.cluster-name)-karpenter" \
+        --attach-policy-arn "arn:aws:iam::$(cat $(results.account-id.path)):policy/KarpenterControllerPolicy-$(params.cluster-name)" \
+        --role-only \
+        --approve
+
+  - name: install-karpenter
+    image: alpine/helm:3.9.0
+    script: |
+      helm repo add karpenter https://charts.karpenter.sh/
+      helm repo update
+      helm upgrade --install --namespace karpenter --create-namespace karpenter karpenter/karpenter \
+        --kubeconfig $(results.kubeconfig.path) \
+        --version $(params.tag) \
+        --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"="arn:aws:iam::$(cat $(results.account-id.path)):role/$(params.cluster-name)-karpenter" \
+        --set clusterName=$(params.cluster-name) \
+        --set clusterEndpoint=$(cat $(results.kubeconfig.path) | grep server | awk '{print $2}') \
```

Comment:
Since Tekton bypasses the image entry point, does this work? Should we #!/usr/bin/env bash the top of this step to make sure?